modular-docs-manual/content/topics/module_anchor-and-file-names-concept.adoc

```diff
@@ -0,0 +1,52 @@
+[#anchor-and-file-names]
+= Anchor Names and File Names
+
+To optimize modular documentation, follow these guidelines for naming module anchors and files.
+
+Anchor names:: Provide an anchor for every module so that it can be identified when reused or cross-referenced. Give the anchor name the same or a similar name to the module heading, separated by underscores, and indicate the module type (`con`, `ref`, `proc`) at the end of the name. If applicable, consider also providing a module sub-type (`create`, `edit`, `delete`) before the module type:
++
+[source]
+----
+[#anchor_name]
+= Module Heading
+
+First sentence of topic.
+----
++
+.Example 1. Concept module
+[source]
+----
+[#guided_decision_tables_con]
+= Guided Decision Tables
+
+The guided decision tables feature works similarly to ...
+----
++
+.Example 2. Procedure module
+[source]
+----
+[#guided_decision_tables_create_proc]
+= Creating Guided Decision Tables
+
+You can use guided decision tables to ...
+----
+
+NOTE: Although you can use other formats for naming anchors, the format defined here is recommended and in some cases required to enable topics to be properly cross-referenced and reused. For details, see xref:reusing-modules[].
+
+For more information about Asciidoc anchors, see the link:http://asciidoctor.org/docs/user-manual/#anchordef[Asciidoctor User Manual].
+
+File names:: Give an informative name separated by dashes and indicate the module type (`con`, `ref`, `proc`) at the end of the name. If applicable, consider also providing a module sub-type (`create`, `edit`, `delete`) right before the module type. Indicating the module type and sub-type at the end of the file name instead of the beginning of the file name results in topics being sorted based on subject matter rather than module type. This assortment can be helpful for managing related content.
```

Comment:
> Indicating the module type and sub-type at the end of the file name instead of the beginning of the file name results in topics being sorted based on subject matter rather than module type.

Not exactly true if you keep anchor IDs close to module titles, especially in procedures (they usually start with a gerund).

Reply:
Say you have two procedures for columns in guided decision tables: one for adding columns, one for removing columns. With your proposal, these file names would be called "adding-guided-decision-table-columns-proc.adoc" and "removing-guided-decision-table-columns-proc.adoc" (I think, if I've understood you). These would then be very far apart from each other in the project's list of alphabetized of topics, and I actually don't see how their matching the HTML output would necessarily be more helpful. Certainly could be if it's an exact match and you're basing your changes on HTML output, but if you're not looking at the HTML and you just see in a JIRA that you need to change everything in the "Guided Decision Table Columns" chapter, then you'd have to go section-by-section in the HTML to confirm the name of each to then go look for the matching modules, scattered around in the adoc file list.

With what I'm saying, these would instead be named "guided-decision-table-columns-add-proc.adoc" and "guided-decision-table-columns-remove-proc.adoc", and true, they don't look anything like the sections in the output, BUT if I want to update anything at all dealing with guided decision tables, regardless or module type, sub-type, or whatever, I can go directly to the set of "guided decision tables" files clumped together in the adoc file list and find the right module type within that set. It's a trade-off, of course, from the alternative to match the output.

I care less about the exact formatting, and thus am happy to nix or change what I've said about sub-types, etc., but my main argument is to keep the general subject matter as the unvarying identifiable content at the start of the file name so that the adoc list is a collection of clumped, related files, not a scattered array of disparate files.

I've done both what you're saying and what I'm proposing, and have personally found what I've proposed much easier and more logical when making updates.

This is only my opinion, and I'm certainly open to your and the group's suggestions, based on everyone's experience. But just wanted to counter to make sure my rationale was clear. Apologies if I didn't understand you correctly.

Reply:
You understood me perfectly, and I agree that the sorting of the files in the directory suffers with this. To me, it is the price to pay for being able to find any file based on the title alone. I do that quite often, but I'd like other team members to weigh in on this with their experience.

A possible solution dawned on me: What if we keep the file names close to the titles, but arrange the files into directories based on subject matter? See example:
```
topics/
  guided-decision-tables/
    adding-guided-decision-table-columns-proc.adoc
    removing-guided-decision-table-columns-proc.adoc
```

I believe this would work pretty well, at least in my head, and I think it satisfies both our requirements.

Reply:
Based on our IRC discussion with @sterobin, I withdraw this suggestion, and will file a follow-up issue where we can resolve this matter somehow. In the context of this PR, consider all my concerns addressed.